cmd/exec_unix_test.go

```diff
@@ -0,0 +1,67 @@
+package cmd
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/chanzuckerberg/aws-oidc/pkg/aws_config_client"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/ini.v1"
+)
+
+func TestAWSEnvPrecedence(t *testing.T) {
+	r := require.New(t)
+
+	config := ini.Empty()
+	acct1Section, err := config.NewSection("profile Account1")
+	r.NoError(err)
+	prevRegionKey1, err := acct1Section.NewKey("region", "dummyregion1")
+	r.NoError(err)
+	prevOutputKey1, err := acct1Section.NewKey("output", "dummyoutput1")
+	r.NoError(err)
+
+	// Next steps: load the profile, load environment variables, see that the environment variables were set
+	err = os.Setenv("AWS_OUTPUT", "dummyoutput2")
```

Comment:
the problem with setting environment variables like this in a test is that you now have side effects (state that might leak to other tests). This, with enough time and complexity, makes future testing hard/brittle and code maintenance more expensive since you now have to be careful regarding the interaction between tests.  

you can probably use a function like https://github.com/chanzuckerberg/aws-oidc/blob/e04e4d1a1afc95fddce782fd80700871a05088a3/pkg/aws_config_client/parse_aws_config_test.go#L15 to reset the env as it was before the test ran. It might make sense to abstract this test func to a function you can re-use in other tests.